File: src/Desafio.Domain/ValueObject/Endereco.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Desafio.Domain.ValueObject
{
    public class Endereco
    {
        public Endereco(string cep, string state, string neighborHood, string street, string service, string city)
        {
            Cep = cep;
            State = state;
            NeighborHood = neighborHood;
            Street = street;
            Service = service;
            City = city;
        }

        public string Cep { get;  set; }
        public string State { get; set; }
        public string City { get; set; }
        public string NeighborHood { get; set; }
        public string Street { get; set; }
        public string Service { get; set; }
    }
}

File: Desafio-FrontEnd/src/app/models/endereco.ts
export class Endereco {
    /**
     *
     */
    constructor(
        cep?: string,
        state?: string,
        city?: string,
        street?: string,
        service?: string,
        neighborhood?: string
    ) {
        this.cep = cep;
        this.state = state;
        this.city = city;
        this.street = street;
        this.service = service;
        this.neighborhood = neighborhood;
    }

    public cep?: string;
    public state?: string;
    public city?: string;
    public street?: string;
    public service?: string;
    public neighborhood?: string;
}